### DISCUSSION TITLE
Why a secret .prisma folder?

### DISCUSSION THREAD

**Discussion by sarink** [prisma/prisma#11475] [Q&A] [answered]
Why are TS types and client options generated in a hidden folder? 

It would be better for `generate` to output a `prisma-types.ts` file that gets checked in to source control.

And/or a constructor options file that gets generated and checked in, maybe used like `const client = new PrismaClient(require('./generated-client-options'));`

The hidden folder seems like a very poor design, as it only causes problems and confusion. What is the benefit/why was this done?

**Comment by austincrim** [comment, marked as answer]
Hey @sarink 👋 ,

You can specify the location for the generated Prisma client in your `schema.prisma` like this:
```prisma
generator client {
  provider = "prisma-client-js"
  output   = "./my-prisma-stuff"
}
```

Generated files are typically not checked into source control, but you can certainly setup your repository however you like.

Prisma generates to `node_modules/.prisma` by default so that it can be imported and treated like a regular package, instead of a generated one. We use the `.prisma` folder instead of `@prisma` to avoid modifying an already downloaded package.

Let me know if I can clarify anything else!

**Comment by sarink** [comment]
hey @austincrim , why do you say generated files are not typically checked in to source control? This seems the opposite from how most other codegen-style tooling works (most notably, apollo and graphql-code-generator). The TS types and the generated code inside `.prisma` is important and extremely specific only to your code, it's not a general-use node module. I find this decision confusing.

**Reply by tlindener** [reply]
The Prisma Schema is the single source of truth for the TS types. Therefore as long as you always rebuild from the same schema you should always get the same types. Seems pointless to include it into the repo since you will have to fix merge conflicts even though a simple regeneration solves it directly.

**Reply by austincrim** [reply]
@sarink You are definitely free to generate your client wherever you want and include it in your repo as you see fit 👍.

**Reply by sarink** [reply]
Thanks for the explanation!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
